core/config/general_config.go

```diff
@@ -185,6 +186,9 @@ type BasicConfig interface {
 type GeneralConfig interface {
 	BasicConfig
 	ValidateDB() error
+	NewLogger() logger.Logger
+	Logger() logger.Logger
+	CloseLogger() error
```

Comment:
I'm worried about these methods being misused/abused in the future. Is it possible to restrict them somehow? e.g. at least define as a separate documented interface that gets embedded here - or not included here at all, if that is possible.

Reply:
@jmank88 Hmmmm... fair point! I'm sure we can get rid of Logger() and CloseLogger(), but barring my suggestion below we will need at least one method to interact with `generalConfig.lggr`. This could be either the `NewLogger()` function we already have or a setter function.

That is unless we want to remove all logger calls from the config, which actually feels like a better alternative? There are two of these at the moment, and both are invoked in calls responsible for lazily generating some data, namely: P2PListenPort(), which generates a random port, and RootDir() which lazily expands the rootdir.

If we did both of those in the config init and returned an error to be logged higher up the stack, we could then break the link between config <> logger entirely.

What do you think of that as a solution?

Reply:
Yes, if that is doable then it sounds better.